src/models/room-state.ts

```diff
@@ -401,6 +414,8 @@ export class RoomState extends TypedEventEmitter<EmittedEvents, EventHandlerMap>
 
                 // assume all our sentinels are now out-of-date
                 this.sentinels = {};
+            } else if (event.getType() === EventType.Marker) {
+                this.emit(RoomStateEvent.Marker, event, setStateOptions);
```

Comment:
When we see a new marker state event, emit an event so we can listen for it in `sync.ts` -> `registerStateListeners` and see whether we need to refresh the timeline